File: csharp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/resueltos/E01_RepasoVariablesIf_Resuelto.cs
/*
OBJETIVO: int nota=7; if >=5 Aprobado else Suspenso.
SOLUCION: ver codigo.
PATRON: Menu do-while con opcion 0 salir y uso de Console.ReadLine().
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/
using System;
class Program
{
    static void Main()
    {
        string? opcion;
        do
        {
            Console.WriteLine();
            Console.WriteLine("=== E01 Repaso Variables If Resuelto ===");
            Console.WriteLine("1. Ejecutar ejemplo del enunciado");
            Console.WriteLine("2. Probar con otra nota");
            Console.WriteLine("0. Salir");
            Console.Write("Elige una opcion: ");
            opcion = Console.ReadLine();
            switch (opcion)
            {
                case "1":
                    MostrarResultadoNota(7);
                    break;
                case "2":
                    Console.Write("Nota: ");
                    int nota = int.Parse(Console.ReadLine() ?? "0");
                    MostrarResultadoNota(nota);
                    break;
                case "0":
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida.");
                    break;
            }
        } while (opcion != "0");
    }
    static void MostrarResultadoNota(int nota)
    {
        if (nota >= 5)
        {
            Console.WriteLine("Aprobado");
        }
        else
        {
            Console.WriteLine("Suspenso");
        }
    }
}

File: cpp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/resueltos/E01_RepasoVariablesIf_Resuelto.cpp
/*
OBJETIVO: int nota=7; if >=5 Aprobado else Suspenso.
SOLUCION: ver codigo.
PATRON: Menu do-while con opcion 0 salir y uso de cin.
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E01 Repaso Variables If Resuelto ===" << endl;
    cout << "1. Ejecutar ejemplo del enunciado" << endl;
    cout << "2. Probar con otra nota" << endl;
    cout << "0. Salir" << endl;
}

void MostrarResultadoNota(int nota)
{
    if (nota >= 5)
    {
        cout << "Aprobado" << endl;
    }
    else
    {
        cout << "Suspenso" << endl;
    }
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                MostrarResultadoNota(7);
                break;
            case 2: {
                int nota;
                cout << "Nota: ";
                cin >> nota;
                MostrarResultadoNota(nota);
                break;
            }
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}
